lib/valueflow.cpp

```diff
@@ -4259,26 +4259,30 @@ struct LifetimeStore {
     }
 };
 
-static bool isOwningVariables(const std::list<Variable>& vars, int depth = 10)
+static bool isOwningVariables(const std::list<Variable>& vars, const std::vector<const Token*>& args, int depth = 10)
 {
     if (depth < 0)
         return false;
-    return vars.empty() || std::all_of(vars.cbegin(), vars.cend(), [&](const Variable& var) {
-        if (var.isReference() || var.isPointer())
-            return false;
+    return std::any_of(vars.cbegin(), vars.cend(), [&](const Variable& var) {
         const ValueType* vt = var.valueType();
         if (vt) {
-            if (vt->pointer > 0)
+            if (std::none_of(args.begin(), args.end(), [vt](const Token* arg) {
+                return arg->valueType() && arg->valueType()->type == vt->type;
+            }))
```

Comment:
I think this check can only be done with pointers. Another user class or container view could still borrow the lifetime.